main.1.tf

```diff
@@ -0,0 +1,110 @@
+variable "region" {
+  default = "us-east-1"
+}
+
+variable "shared_credentials_file" {
+  default = "/home/ec2-user/.aws/credentials"
+}
+
+variable "profile" {
+  default = "terraform"
+}
+
+provider "aws" {
+  region                  = "${var.region}"
+  shared_credentials_file = "${var.shared_credentials_file}"
+  profile                 = "${var.profile}"
+}
+
+
+resource "aws_instance" "web" {
+  ami           = "ami-062f7200baf2fa504"
+  instance_type = "t2.micro"
+
+  tags = {
+    Name = "HelloWorld"
+  }
+}
+
+resource "aws_launch_template" "spot-demo-lt" {
+  name = "spot-demo-lt"
+
+  block_device_mappings {
+    device_name = "/dev/sda1"
+
+    ebs {
+      volume_size = 20
+    }
+  }
+
+  ebs_optimized = true
+
+  image_id = "ami-062f7200baf2fa504"
+
+
+  instance_type = "t3.micro"
+
+  key_name = "awsajp_keypair"
+
+
+  monitoring {
+    enabled = true
+  }
+
+  network_interfaces {
+    associate_public_ip_address = true
+  }
+
+  placement {
+    availability_zone = "us-east-1a,us-east-1b,us-east-1c,us-east-1d,us-east-1e,us-east-1f"
+  }
+
+  vpc_security_group_ids = ["sg-4f3f0d1e"]
```

Comment:
Making reference to internal, pre-created security groups ??? Not something that we should be doing with terraform, right ? Create the security group and refer to it